### PR TITLE
feat: implement sortIcon on Table

### LINKED ISSUE #19561: Table component doesn't allow custom sort icons
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
Our team is using Ant tables, but we would like to change the sort icon to be an arrow instead of a triangle.  We were wondering if we could send the sort icon via a prop type, but we could not find any prop type that satisfies this.

### What does the proposed API look like?
`<Table sortUpIcon={<MyIconAscend />}  sortDownIcon={<MyIconDescend />} sortNoneIcon={<MyIconNoSort />} />`

### PR DESCRIPTION
**Pull request by sawadyecma**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close https://github.com/ant-design/ant-design/issues/19561

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Add support for custom sort icon of table column |
| 🇨🇳 Chinese | 添加对表格列自定义排序图标的支持 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at fa68788</samp>

This pull request adds a new feature to the table component: custom sort icons for columns. It also improves the code quality by refactoring the `useSorter` hook and the `ColumnType` interface.

### <samp>🤖 Generated by Copilot at fa68788</samp>

* Add support for custom sort icon of table column ([link](https://github.com/ant-design/ant-design/pull/42498/files?diff=unified&w=0#diff-33a34c8ef463c5e1cad14a8046c87aa47265d4927644753c226a65211511dc41L130-R164), [link](https://github.com/ant-design/ant-design/pull/42498/files?diff=unified&w=0#diff-3fee134a410a783dfd5f6f89b1a8f7e7eb1ba8e332aeaf4cafdefc4c723088e9R122))
* Simplify sorter element rendering in `useSorter` hook ([link](https://github.com/ant-design/ant-design/pull/42498/files?diff=unified&w=0#diff-33a34c8ef463c5e1cad14a8046c87aa47265d4927644753c226a65211511dc41L162-R183))

### CODE REVIEW

File: components/table/interface.ts [hunk 1]
```diff
@@ -119,6 +119,7 @@ export interface ColumnType<RecordType> extends Omit<RcColumnType<RecordType>, '
   sortOrder?: SortOrder;
   defaultSortOrder?: SortOrder;
   sortDirections?: SortOrder[];
+  sortIcon?: React.ReactNode | ((sortOrder: SortOrder) => React.ReactNode);
```
Comment (afc163) [inline components/table/interface.ts:122]: Should we drop `React.ReactNode` type? Is `((sortOrder: SortOrder) => React.ReactNode)` enough already?

Comment (sawadyecma) [reply components/table/interface.ts:122]: I agree with you and gonna erase it. Thank you so much.

Comment (sawadyecma) [reply components/table/interface.ts]: I've fixed it.
https://github.com/ant-design/ant-design/pull/42498/commits/bc4779aea8815c5e152d83704b0d7d4634b8cdbc

**Comment by sawadyecma** [comment]
Can someone write changelog, please?

File: components/table/hooks/useSorter.tsx [hunk 2]
```diff
@@ -127,20 +127,39 @@ function injectSorter<RecordType>(
...
-        <CaretDownOutlined
-          className={classNames(`${prefixCls}-column-sorter-down`, {
-            active: sorterOrder === DESCEND,
-          })}
-        />
-      );
+
+      let sorter: React.ReactNode;
+      if (column.sortIcon) {
+        sorter = column.sortIcon(sorterOrder);
```
Comment (afc163) [inline components/table/hooks/useSorter.tsx:133]:
```suggestion
        sorter = column.sortIcon({ sorterOrder });
```

Incase that we could add more argument in future.

Comment (sawadyecma) [reply components/table/hooks/useSorter.tsx]: I've fixed it.
https://github.com/ant-design/ant-design/pull/42498/commits/2935cdb0db0a2c72bf2344392a339c2ef3c24a33

**Comment by afc163** [comment]
Could you send this to feature branch?

**Comment by afc163** [comment]
The documentation should be updated too.

**Review by afc163 — APPROVED** [review]

**Comment by li-jia-nan** [comment]
文档里是不是没加版本号？

**Comment by li-jia-nan** [comment]
- #42518

**Comment by vitalijalbu** [comment]
How to use this? I tried but without success:
@afc163 thanks
```
<Table
sortIcon={({ sortOrder }) => 
  sortOrder === 'ascend' ? <IconSelector size={18} className="text-gray-400" /> 
  : sortOrder === 'descend' ? <IconSelector size={18} className="text-gray-600" /> 
  : null
}/>
```

**Pull request merged**